server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/data/StarTreeDataWriter.java

```diff
@@ -66,21 +63,26 @@ private static void writeStarTreeNodes(IndexOutput output, InMemoryTreeNode root
         while (!queue.isEmpty()) {
             InMemoryTreeNode node = queue.remove();
 
-            if (node.children == null || node.children.isEmpty()) {
+            if (!node.hasChild()) {
                 writeStarTreeNode(output, node, ALL, ALL);
             } else {
 
-                // Sort all children nodes based on dimension value
-                List<InMemoryTreeNode> sortedChildren = new ArrayList<>(node.children.values());
-                sortedChildren.sort(
-                    Comparator.comparingInt(InMemoryTreeNode::getNodeType).thenComparingLong(InMemoryTreeNode::getDimensionValue)
-                );
-
+                int totalNumberOfChildren = 0;
                 int firstChildId = currentNodeId + queue.size() + 1;
-                int lastChildId = firstChildId + sortedChildren.size() - 1;
+
+                if (node.getChildStarNode() != null) {
+                    totalNumberOfChildren++;
```

Comment:
do we need totalNumberOfChildren ? 
```
 
 int lastChildId = firstChildId + node.getChildren().values().size()  ; ?

```

Reply:
We maintain the first child id and last child id for file formats. totalNumberOfChildren tracks star child as well which ```node.getChildren().values().size()``` will not